grakn-engine/src/main/java/ai/grakn/engine/task/postprocessing/RedisIndexStorage.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.engine.task.postprocessing;
+
+import ai.grakn.Keyspace;
+import ai.grakn.concept.ConceptId;
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.annotations.VisibleForTesting;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.Response;
+import redis.clients.jedis.Transaction;
+import redis.clients.util.Pool;
+
+import javax.annotation.Nullable;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ *
+ * <p>
+ *     Stores a list of indices and vertex ids representing those indices which need to be post processed
+ * </p>
+ *
+ * @author fppt
+ */
+public class RedisIndexStorage extends RedisStorage {
```

Comment:
You should totally try the Intellij thing: Refactor -> Replace Inheritance With Delegation

Not 100% saying it will be better in this case, but maybeee?